homeassistant/components/playstation_network/coordinator.py

```diff
@@ -0,0 +1,61 @@
+"""Coordinator for the PlayStation Network Integration."""
+
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+
+from psnawp_api.core.psnawp_exceptions import PSNAWPAuthenticationError
+from psnawp_api.models.user import User
+from psnawp_api.psn import PlaystationNetwork, PlaystationNetworkData
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+type PlaystationNetworkConfigEntry = ConfigEntry[PlaystationNetworkCoordinator]
+
+
+class PlaystationNetworkCoordinator(DataUpdateCoordinator[PlaystationNetworkData]):
+    """Data update coordinator for PSN."""
+
+    config_entry: PlaystationNetworkConfigEntry
+
+    def __init__(
+        self, hass: HomeAssistant, psn: PlaystationNetwork, user: User
+    ) -> None:
+        """Initialize the Coordinator."""
+        super().__init__(
+            hass,
+            name=DOMAIN,
+            logger=_LOGGER,
+            update_interval=timedelta(seconds=30),
+        )
+
+        self.hass = hass
+        self.user: User = user
+        self.psn: PlaystationNetwork = psn
```

Comment:
```suggestion
        self.user = user
        self.psn = psn
```
Type can be inferred